### ISSUE TITLE
Add non-generic constants' value to metadata and stop encoding their MIR

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#83011]
I think it is cheaper to store the evaluated constant in crate metadata than to store its entire MIR. The only reason we encode the MIR right now is so that downstream crates can evaluate the constant again.

cc @rust-lang/wg-const-eval any problems with this that come to mind? We may be able to do this for promoteds in non-generic functions, too. Paired with polymorphization we can even do that for polymorphic constants at some point (as long as the value does not depend on the generic parameters).

**Comment by nagisa** [comment]
> I think it is cheaper to store the evaluated constant in crate metadata than to store its entire MIR. The only reason we encode the MIR right now is so that downstream crates can evaluate the constant again.

Would this not hold true for e.g. arrays such as `const BAR: ... = [const_fn(); get_size()]`? Storing the description on how to generate a large array definitely sounds cheaper (storage wise) than storing the value itself (which can get pretty large)

**Comment by oli-obk** [comment]
That's true. Maybe we should just do this for scalar constants then. These are easy to detect and will always be a gain.

**Comment by nagisa** [comment]
Yeah, that would make sense to me.
